src/NewTools-FileBrowser/StFileBrowserTextBasedContentPreviewer.class.st

```diff
@@ -0,0 +1,63 @@
+"
+I'm a previewer use to display the text content of a file 
+"
+Class {
+	#name : 'StFileBrowserTextBasedContentPreviewer',
+	#superclass : 'StFileBrowserAbstractPreviewer',
+	#category : 'NewTools-FileBrowser-Previewers',
+	#package : 'NewTools-FileBrowser',
+	#tag : 'Previewers'
+}
+
+{ #category : 'testing' }
+StFileBrowserTextBasedContentPreviewer >> canBeUsedOn: aFileReferenceOrString [
+	"Answer <true> if aFileReferenceOrString is a non-empty text file"
+	| fileRef |
+	
+	fileRef := aFileReferenceOrString asFileReference.
+	^ fileRef isFile and: [ fileRef size ~= 0 and: [ self isTextStream: fileRef ] ]
+]
+
+{ #category : 'testing' }
+StFileBrowserTextBasedContentPreviewer >> isTextStream: aZnCharacterReadStream [ 
+	"Answer <true> if parameter represents a text stream"
+	
+	aZnCharacterReadStream readStreamDo: [ :s | 
+		[ s next: self numberOfBytesToCheckForBinaryOrText ]
+		   on: ZnInvalidUTF8
+		   do: [ ^ false ] ].
+	^ true
+]
+
+{ #category : 'testing' }
+StFileBrowserTextBasedContentPreviewer >> matchingMimeType: aZnMimeType [
+	"return a Boolean to indicate if element matches"
+	^ true
+]
+
+{ #category : 'private' }
+StFileBrowserTextBasedContentPreviewer >> numberOfBytesToCheckForBinaryOrText [
+	
+	^ 100
+]
+
+{ #category : 'as yet unclassified' }
```

Comment:
```suggestion
{ #category : 'accessing' }
```

Reply:
@hernanmd can you rewrite it so that we push it then?

Reply:
Ok, done